smsclient_core/models/sms_gateway.py

```diff
@@ -0,0 +1,220 @@
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        help='The sms class: flash(0), phone display(1), SIM(2), toolkit(3)')
+    deferred = fields.Integer(
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        help='The time -in minute(s)- to wait before sending the message.')
+    priority = fields.Selection(
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        selection=PRIORITY_LIST,
+        help='The priority of the message ')
+    coding = fields.Selection([
+        ('1', '7 bit'),
+        ('2', 'Unicode')
+        ], readonly=True,
+        states={'draft': [('readonly', False)]},
+        help='The sms coding: 1 for 7 bit or 2 for unicode')
+    tag = fields.Char(
+        size=256,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        help='An optional tag')
+    nostop = fields.Boolean(
+        string='NoStop',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        help='Do not display STOP clause in the message, this requires that'
+             'this is not an advertising message.')
+    partner_id = fields.Many2one(
+        'res.partner',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        string='Partner')
+    company_id = fields.Many2one(
+        comodel_name='res.company',
+        readonly=True,
+        states={'draft': [('readonly', False)]})
+
+    @api.onchange('partner_id')
+    def onchange_partner_id(self):
+        self.mobile = self.partner_id.mobile
+
+    @api.multi
+    def send(self):
+        for sms in self:
+            if not hasattr(sms, "_send_%s" % sms.gateway_id.method):
+                raise UserError(_("No method gateway selected"))
+            if not sms.gateway_id._check_permissions():
+                sms.write(
+                    {'error': 'no permission on gateway', 'state': 'error'})
+                sms._cr.commit()
+                return False
+            if sms.gateway_id.char_limit <= len(sms.message):
+                sms.write({
+                    'state': 'error',
+                    'error': _(
+                        'Size of SMS should not be more than %s characters '
+                        % sms.gateway_id.char_limit)
+                })
+                sms._cr.commit()
+                raise UserError(
+                    _('Size of SMS should not be more than %s characters set '
+                      'in gateway options' % sms.gateway_id.char_limit))
+            else:
+                try:
+                    with sms._cr.savepoint():
+                        getattr(sms, "_send_%s" % sms.gateway_id.method)()
+                        sms.write({'state': 'sent', 'error': ''})
+                except Exception, e:
+                    _logger.error('Failed to send sms %s', e)
+                    sms.write({'error': e, 'state': 'error'})
+                sms._cr.commit()
```

Comment:
[Please do not commit the transaction](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#never-commit-the-transaction)